File: String/leetcode_38.cpp

// Problem Link ====>>https://leetcode.com/problems/count-and-say/description/

#include<bits/stdc++.h>
using namespace std;

class Solution{
    public:
    string countAndSay(int n){
        switch(n){
            case 1:
                return "1";
            case 2:
                return "11";
        }

        string str = "11";
        int counter = 1;
        string temp = "";

        for (int i = 3; i <= n; ++i){
            str += "&";
            for (int j = 1; j < str.length(); ++j){
                if(str[j] != str[j-1]){
                    temp += to_string(counter) + str[j - 1];
                    counter = 1;
                } else {
                    ++counter;
                }
            }
            counter = 1;
            str = move(temp);
            temp.clear();
        }
        return str;
    }
};

int main()
{
    int n;
    cout << "Enter the value of n: ";
    cin >> n;

    Solution solution;
    string result = solution.countAndSay(n);

    cout << "The " << n << "th term in the count-and-say sequence is: " << result << endl;

    return 0;
}

File: String/leetcode_38.py
# Problem Link ====>>https://leetcode.com/problems/count-and-say/description/

class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        if n == 2:
            return "11"
        
        prev = "11"
        for _ in range(2, n):
            curr = ""
            count = 1
            for i in range(len(prev)):
                if i == 0 or prev[i] == prev[i-1]:
                    count += 1
                else:
                    curr += str(count) + prev[i-1]
                    count = 1
            curr += str(count) + prev[-1]
            prev = curr
        return prev

# Example usage
solution = Solution()
result = solution.countAndSay(4)
print(f"The 4th term in the count-and-say sequence is: {result}")
